### PR TITLE
closes #23565: expire remote wallpapers but let users keep selected

### LINKED ISSUE #23565: Add expiration date for wallpapers
Certain remote wallpapers will have an expiration date tied to them. When wallpapers expire, we want to delete any unused copies of them from user's devices. 

The exception is that if a user has an expired wallpaper selected, they can continue to use it.

### PR DESCRIPTION
**Pull request by MatthewTighe**
### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Review by Amejia481 — APPROVED** [review]
👏🏽  awesome work !!!

File: app/src/main/java/org/mozilla/fenix/wallpapers/WallpaperManager.kt [hunk 1]
```diff
@@ -95,33 +97,43 @@ class WallpaperManager(
         }
     }
 
+    private fun filterExpiredRemoteWallpapers(wallpaper: Wallpaper): Boolean = when (wallpaper) {
+        is Wallpaper.Remote -> {
+            val notExpired = wallpaper.expirationDate?.let { Date().before(it) } ?: true
+            notExpired || wallpaper.name == settings.currentWallpaper
```
Comment (jonalmeida) [inline app/src/main/java/org/mozilla/fenix/wallpapers/WallpaperManager.kt:103]: Drive-by 🚗 : Out of curiosity, if had a remote wallpaper selected at some point, but is no longer selected, would that disappear?

Comment (Amejia481) [reply app/src/main/java/org/mozilla/fenix/wallpapers/WallpaperManager.kt:103]: Yes, If the wallpaper is expired it will disappear.

Comment (MatthewTighe) [reply app/src/main/java/org/mozilla/fenix/wallpapers/WallpaperManager.kt:103]: That's currently the spec. We'll hopefully add some UX around that to make it a little more user-friendly soon. It's easy to imagine improving the persistence layer eventually to allow people to keep them for longer as well

Comment (jonalmeida) [reply app/src/main/java/org/mozilla/fenix/wallpapers/WallpaperManager.kt:103]: Sounds good, thanks that clarifying!

**Pull request merged**